### PR TITLE
feat: created challenge card for the grid with hover functionality

### LINKED ISSUE #4: Challenge grid card component
## Description
Card which shows challenge details, transparent on hover. Use Text and Icon components to make.
image

### Reference page
image

NOTE: Use **storybook** to preview your component. Ideally **do not create pages** to test your component. If you do make on, remember to **delete the page before pushing**. Refer src/components/text.tsx and src/stories/text.stories.tsx for examples.

### PR DESCRIPTION
**Pull request by Swik03**
hopefully fixes #4 

The card is using a react-icon component and the Text component directly, and hovering changes the opacity without any transition effect.

### CODE REVIEW

File: src/components/challengeCard.tsx [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+              style={{ fontSize: "16px" }}
+            >
+              {type}
+            </Text>
+            <Text
+              variant="secondary"
+              glow="secondary"
+              style={{ fontSize: "18px", fontWeight: "bold" }}
+            >
+              <br></br>
```
Comment (ThEditor) [inline src/components/challengeCard.tsx]: Can you explain these `<br>` tags?

File: src/components/challengeCard.tsx [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+        onMouseEnter={() => setIsHovered(true)}
+        onMouseLeave={() => setIsHovered(false)}
+      >
+        <div style={cardStyle}>
+          <div style={contentStyle}>
+            <BsNut size={52} style={{ transform: "rotate(90deg)" }} />
+            <Text
+              variant="secondary"
+              glow="secondary"
+              style={{ fontSize: "24px", fontWeight: "bold" }}
```
Comment (ThEditor) [inline src/components/challengeCard.tsx]: Utilize tailwindcss instead (run through some AI model or smthng lol)

File: src/components/challengeCard.tsx [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+            >
+              <br></br>
+              {title}
+            </Text>
+            <Text
+              variant="secondary"
+              glow="secondary"
+              style={{ fontSize: "16px" }}
```
Comment (ThEditor) [inline src/components/challengeCard.tsx]: Utilize tailwindcss

File: src/components/challengeCard.tsx [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+            >
+              {type}
+            </Text>
+            <Text
+              variant="secondary"
+              glow="secondary"
+              style={{ fontSize: "18px", fontWeight: "bold" }}
```
Comment (ThEditor) [inline src/components/challengeCard.tsx]: Utilize tailwindcss

File: src/components/challengeCard.tsx [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+              glow="secondary"
+              style={{ fontSize: "18px" }}
+            >
+              <br></br>
+              {description}
+            </Text>
+            <Text
+              variant="secondary"
+              glow="secondary"
+              style={{ fontSize: "16px", fontWeight: "bold" }}
```
Comment (ThEditor) [inline src/components/challengeCard.tsx]: Utilize tailwindcss

File: src/components/challengeCard.tsx [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+              glow="secondary"
+              style={{ fontSize: "18px", fontWeight: "bold" }}
+            >
+              <br></br>
+              {points} {difficulty}
+            </Text>
+            <Text
+              variant="secondary"
+              glow="secondary"
+              style={{ fontSize: "18px" }}
```
Comment (ThEditor) [inline src/components/challengeCard.tsx]: Utilize tailwindcss

File: src/components/challengeCard.tsx [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+            </Text>
+          </div>
+        </div>
+      </div>
+    );
+  }
+};
+
+// Styles (using inline for simplicity)
+const cardStyle: React.CSSProperties = {
```
Comment (ThEditor) [inline src/components/challengeCard.tsx]: Utilize tailwindcss

File: src/components/challengeCard.tsx [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+  borderRadius: "16px",
+  width: "520px",
+  boxShadow: "0 4px 10px rgba(0, 0, 0, 0.2)",
+  transition: "background-color 0.3s ease", // Hover effect
+  cursor: "pointer",
+  color: "#000000",
+  marginBottom: "20px",
+};
+
+const contentStyle: React.CSSProperties = {
```
Comment (ThEditor) [inline src/components/challengeCard.tsx]: Utilize tailwindcss

**Review by ThEditor — CHANGES_REQUESTED** [review]
These changes are not mandatory but will be nice to have since I would want the code to follow a similar style. Thanks for contributing though :D

File: src/components/challengeCard.tsx (hunk 1 above)
Comment (Swik03) [reply src/components/challengeCard.tsx]: It was a dirty fix to make better margins between the text lol

File: src/components/challengeCard.tsx (hunk 2 above)
Comment (Swik03) [reply src/components/challengeCard.tsx]: Rewritten it now lol

**Comment by Swik03** [comment]
I think I've addressed the changes, refactored and removed the glow :D

**Review by ThEditor — APPROVED** [review]
lgtm

File: src/stories/challengeCard.stories.tsx [hunk 9]
```diff
@@ -0,0 +1,53 @@
+import challengeCard from "~/components/challengeCard";
+import type { Meta, StoryObj } from "@storybook/react";
+
+const meta: Meta<typeof challengeCard> = {
+  title: "Text",
```
Comment (ThEditor) [inline src/stories/challengeCard.stories.tsx]: rename to `ChallengeCard`

File: src/components/challengeCard.tsx [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+  difficulty: string;
+  solves: number;
+}> = ({ title, type, description, points, difficulty, solves = 0 }) => {
+  const [isHovered, setIsHovered] = useState(false);
+  {
+    return (
+      <div
+        className={`p-6 rounded-lg shadow-lg transition-opacity duration-300 cursor-pointer mb-5 ${isHovered ? "opacity-80" : "opacity-100"
+          }`}
+        onMouseEnter={() => setIsHovered(true)}
```
Comment (ThEditor) [inline src/components/challengeCard.tsx]: no need of explicit hover event
do `hover:bg-transparent`

File: src/components/challengeCard.tsx [hunk 11]
```diff
@@ -0,0 +1,52 @@
...
+  description: string;
+  points: number;
+  difficulty: string;
+  solves: number;
+}> = ({ title, type, description, points, difficulty, solves = 0 }) => {
+  const [isHovered, setIsHovered] = useState(false);
+  {
+    return (
+      <div
+        className={`p-6 rounded-lg shadow-lg transition-opacity duration-300 cursor-pointer mb-5 ${isHovered ? "opacity-80" : "opacity-100"
```
Comment (ThEditor) [inline src/components/challengeCard.tsx]: `shadow-lg` does not reflect the actual design, remove shadow

Comment (ThEditor) [inline src/components/challengeCard.tsx]: remove `transition-opacity`
add `transition-bg duration-200 ease-in-out`

File: src/components/challengeCard.tsx [hunk 12]
```diff
@@ -0,0 +1,52 @@
...
+          }`}
+        onMouseEnter={() => setIsHovered(true)}
+        onMouseLeave={() => setIsHovered(false)}
+        style={{ backgroundColor: "rgba(30, 30, 33, 1)" }}
```
Comment (ThEditor) [inline src/components/challengeCard.tsx]: inline style used, use tailwindcss
`bg-[#2D2D2D80]`

**Review by ThEditor — CHANGES_REQUESTED** [review]

File: src/stories/challengeCard.stories.tsx (hunk 9 above)
Comment (Swik03) [reply src/stories/challengeCard.stories.tsx]: done

File: src/components/challengeCard.tsx (hunk 10 above)
Comment (Swik03) [reply src/components/challengeCard.tsx]: done

File: src/components/challengeCard.tsx (hunk 11 above)
Comment (Swik03) [reply src/components/challengeCard.tsx]: done

Comment (Swik03) [reply src/components/challengeCard.tsx]: done

File: src/components/challengeCard.tsx (hunk 12 above)
Comment (Swik03) [reply src/components/challengeCard.tsx]: done

File: src/components/challengeCard.tsx [hunk 13]
```diff
@@ -0,0 +1,44 @@
...
+  return (
+    <div
+      className={`bg-[#2D2D2D80] hover:bg-transparent p-6 rounded-lg transition-bg duration-200 ease-in-out cursor-pointer mb-5`}
+    >
+      <div className="flex flex-col">
+        <BsNut size={52} className="rotate-90" />
+        <Text className="text-2xl font-bold mt-4 mb-2" variant="secondary" glow="none">
+          {title}
+        </Text>
+        <Text className="text-sm" variant="secondary" glow="none">
```
Comment (ThEditor) [inline src/components/challengeCard.tsx]: `text-white` required on all Text components used here
waiting on @Sasikuttan2163 to add `white` variant to Text component
once that's done you can make these changes

**Review by ThEditor — APPROVED** [review]

File: src/components/challengeCard.tsx (hunk 13 above)
Comment (Sasikuttan2163) [reply src/components/challengeCard.tsx]: `text-white` has been added @Swik03 you can go ahead and change the variant :D

Comment (Swik03) [reply src/components/challengeCard.tsx]: done! commit history's a bit messy though lol

**Comment by ThEditor** [comment]
lgtm, thanks for contributing

**Pull request merged**